### PR TITLE
wat: completes coverage except TODOs

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

File: wasm/wat/parser.go [hunk 1]
```diff
@@ -230,21 +228,31 @@ func (p *moduleParser) parseImport(tok tokenType, tokenBytes []byte, _, _ uint32
 			return fmt.Errorf("redundant name: %s", name)
 		}
 	case tokenLParen: // start fields, ex. (func
+		// Err if there's a second description. Ex. (import "" "" (func) (func))
+		if uint32(len(p.module.importFuncs)) > p.currentImportIndex {
+			return p.unexpectedToken(tok, tokenBytes)
+		}
+		// Err if there's are not enough names when we reach a description. Ex. (import func())
```
Comment (codefromthecrypt) [inline wasm/wat/parser.go:235]:
```suggestion
		// Err if there are not enough names when we reach a description. Ex. (import func())
```

File: wasm/wat/parser.go [hunk 2]
```diff
@@ -230,21 +228,31 @@ func (p *moduleParser) parseImport(tok tokenType, tokenBytes []byte, _, _ uint32
...
-			return errors.New("missing description field") // Ex. (module "Math" "Pi")
+			return errors.New("missing description field") // Ex. missing (func): (import "Math" "Pi")
 		}
 
-		// multiple imports are allowed, so advance in case there's a next.
+		// Multiple imports are allowed, so advance in case there's a next.
 		p.currentImportIndex++
 
-		return p.endField()
+		// Reset parsing state: done late to help give correct error messages if there are multiple descriptions.
```
Comment (codefromthecrypt) [inline wasm/wat/parser.go:253]:
```suggestion
		// Reset parsing state: this is late to help give correct error messages on multiple descriptions.
```

**Review by codefromthecrypt — COMMENTED** [review]
typos

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the look, compadre

**Pull request merged**